pkg/client/client.go

```diff
@@ -431,54 +432,102 @@ func (client *DataSetClient) logStatistics() {
 
 	// if nothing was processed, do not log statistics
 	if processingInSec <= 0 {
-		return
+		return nil
 	}
 
 	// log buffer stats
 	bProcessed := client.buffersProcessed.Load()
 	bEnqueued := client.buffersEnqueued.Load()
 	bDropped := client.buffersDropped.Load()
 	bBroken := client.buffersBroken.Load()
-	client.Logger.Info(
-		"Buffers' Queue Stats:",
-		zap.Uint64("processed", bProcessed),
-		zap.Uint64("enqueued", bEnqueued),
-		zap.Uint64("dropped", bDropped),
-		zap.Uint64("broken", bBroken),
-		zap.Uint64("waiting", bEnqueued-bProcessed-bDropped-bBroken),
-		zap.Float64("processingS", processingInSec),
-	)
+
+	buffersStats := QueueStats{
+		bProcessed,
+		bEnqueued,
+		bDropped,
+		bBroken,
+		bEnqueued - bProcessed - bDropped - bBroken,
+		processingInSec,
+	}
 
 	// log events stats
 	eProcessed := client.eventsProcessed.Load()
 	eEnqueued := client.eventsEnqueued.Load()
 	eDropped := client.eventsDropped.Load()
 	eBroken := client.eventsBroken.Load()
-	client.Logger.Info(
-		"Events' Queue Stats:",
-		zap.Uint64("processed", eProcessed),
-		zap.Uint64("enqueued", eEnqueued),
-		zap.Uint64("dropped", eDropped),
-		zap.Uint64("broken", eBroken),
-		zap.Uint64("waiting", eEnqueued-eProcessed-eDropped-eBroken),
-		zap.Float64("processingS", processingInSec),
-	)
+
+	eventsStats := QueueStats{
+		eProcessed,
+		eEnqueued,
+		eDropped,
+		eBroken,
+		eEnqueued - eProcessed - eDropped - eBroken,
+		processingInSec,
+	}
 
 	// log transferred stats
 	bAPISent := float64(client.bytesAPISent.Load())
 	bAPIAccepted := float64(client.bytesAPIAccepted.Load())
 	throughput := bAPIAccepted / mb / processingInSec
 	successRate := (bAPIAccepted + 1) / (bAPISent + 1)
 	perBuffer := (bAPIAccepted) / float64(bProcessed)
+	transferStats := TransferStats{
+		bAPISent / mb,
+		bAPIAccepted / mb,
+		throughput,
+		perBuffer / mb,
+		successRate,
+		processingInSec,
+		processingDur,
+	}
+
+	return &Statistics{
```

Comment:
Yeah, I think that's a nicer / better API. Thanks.

Reply:
My plan was to fix it long time ago but I didn't have the opportunity. :)